### PR TITLE
Migrate from Nose to pytest

### LINKED ISSUE #626: ship a py.test plugin for hy
see if this is doable

### PR DESCRIPTION
**Pull request by Kodiologist**
Closes #626. (We're not providing a plugin, but the conftest.py shows how to collect Hy tests with pytest.)

### CODE REVIEW

**Comment by Kodiologist** [comment]
Blimey, it actually works!

**Comment by Kodiologist** [comment]
The hack in conftest.py can be removed once pytest-dev/pytest#2374 is in a released version of pytest.

**Comment by Kodiologist** [comment]
Also closes #72, since we're abandoning Nose.

File: tests/native_tests/core.hy [hunk 1]
```diff
@@ -41,7 +41,7 @@
          (except [TypeError] True)
          (else (assert False)))))
 
-(defn test-coll? []
+(defn test-collq []
```
Comment (kirbyfan64) [inline tests/native_tests/core.hy]: *sigh* We *really* need to work on our mangling...

FWIW pytest lets you [override the test prefix](https://docs.pytest.org/en/latest/customize.html#confval-python_functions). Maybe it'd be worth it to do that with something like:

```ini
[pytest]
# *insert comment lamenting state of world, etc.*
python_function = test_* is_test_*
```

Although it's a hack:

- It's not much more hacky than a questionable (haha!) `q` suffix on every single test function.
- It's a bit less weird to read. (random person looking through code: *Why are there random `q`s everywhere?*)
- The test names will make more sense. This is testing `coll?`, not `collq`...

Comment (Kodiologist) [inline tests/native_tests/core.hy]: Okay, changed.

**Comment by Kodiologist** [comment]
Thanks for reviewing and merrging. Please don't squash when a PR is split into logically distinct commits. I've going to redo the merge.

**Pull request merged**